### ISSUE TITLE
Not compiling: gsl missing?

### ISSUE THREAD

**Issue by ufeindt** [mscrim/MVBulkFlow#1]
Hi!

First of all, thanks for making your code publicly available.
I tried compiling your code on a couple of machines, both running Ubuntu but it doesn't work.
As the readme says, I'm just running make in order to do that. 

I ran into a few problems that I could fix myself (through googling; I know only little C++):
The first problem is the option "-03" (with a zero), which does not seem to exist but "-O3" (with the letter O) does, so I changed the FFLAGS value in the Makefile.
Then several functions were not defined in the scope, namely exit, strcmp and sort. This seems to be a LInux problem. Including <cstdlib>, <string.h> and <algorithm> solved that.
However, I still cannot compile. I have tried the alternative LFLAGS value. It seems that in MV.cpp and MLE.cpp like some files are expect in a directory called "gsl/".

Here is the output:
`
g++ -lgsl -lgslcblas usefulfuncs.o cosmology.o MV.o MLE.o MV_bulkflow.o -o MV_bulkflow
MV.o: In function`MV_Mij(double**, double**, int, double**)':
MV.cpp:(.text+0x1ab7): undefined reference to `gsl_matrix_alloc'
MV.cpp:(.text+0x1b2d): undefined reference to`gsl_matrix_set'
MV.cpp:(.text+0x1b52): undefined reference to `gsl_matrix_alloc'
MV.cpp:(.text+0x1b5f): undefined reference to`gsl_permutation_alloc'
MV.cpp:(.text+0x1b77): undefined reference to `gsl_linalg_LU_decomp'
MV.cpp:(.text+0x1b85): undefined reference to`gsl_linalg_LU_invert'
MV.cpp:(.text+0x1b95): undefined reference to `gsl_matrix_get'
MV.o: In function`MV_Gnm(double**, double**, double**, int, double_, double__, double_, int, double, double_, double_)':
MV.cpp:(.text+0x2ab2): undefined reference to `gsl_matrix_alloc'
MV.cpp:(.text+0x2ae0): undefined reference to`gsl_matrix_set'
MV.cpp:(.text+0x2b03): undefined reference to `gsl_matrix_alloc'
MV.cpp:(.text+0x2b0e): undefined reference to`gsl_permutation_alloc'
MV.cpp:(.text+0x2b21): undefined reference to `gsl_linalg_LU_decomp'
MV.cpp:(.text+0x2b2f): undefined reference to`gsl_linalg_LU_invert'
MV.cpp:(.text+0x2b5a): undefined reference to `gsl_matrix_get'
MV.cpp:(.text+0x2d77): undefined reference to`gsl_matrix_alloc'
MV.cpp:(.text+0x2d8f): undefined reference to `gsl_matrix_alloc'
MV.cpp:(.text+0x2d9a): undefined reference to`gsl_permutation_alloc'
MV.cpp:(.text+0x2dad): undefined reference to `gsl_linalg_LU_decomp'
MV.cpp:(.text+0x2dbb): undefined reference to`gsl_linalg_LU_invert'
MLE.o: In function `MLE_BF(int, double**, double*, double, double*, double*, double*, double&, double&, double&, double*, char*, bool, bool)':
MLE.cpp:(.text+0xbc0): undefined reference to`gsl_matrix_alloc'
MLE.cpp:(.text+0xc79): undefined reference to `gsl_matrix_set'
MLE.cpp:(.text+0xcab): undefined reference to`gsl_matrix_alloc'
MLE.cpp:(.text+0xcbd): undefined reference to `gsl_permutation_alloc'
MLE.cpp:(.text+0xcd2): undefined reference to`gsl_linalg_LU_decomp'
MLE.cpp:(.text+0xce5): undefined reference to `gsl_linalg_LU_invert'
MLE.cpp:(.text+0xd57): undefined reference to`gsl_matrix_get'
MLE.cpp:(.text+0x137b): undefined reference to `gsl_matrix_get'
MLE.cpp:(.text+0x1390): undefined reference to`gsl_matrix_get'
MLE.cpp:(.text+0x13a2): undefined reference to `gsl_matrix_get'
MLE.cpp:(.text+0x144a): undefined reference to`gsl_matrix_get'
MLE.o:MLE.cpp:(.text+0x1462): more undefined references to `gsl_matrix_get' follow
MLE.o: In function`MLE_BF(int, double**, double_, double, double_, double_, double_, double&, double&, double&, double_, char_, bool, bool)':
MLE.cpp:(.text+0x1642): undefined reference to `gsl_matrix_alloc'
MLE.cpp:(.text+0x1654): undefined reference to`gsl_permutation_alloc'
MLE.cpp:(.text+0x1667): undefined reference to `gsl_linalg_LU_decomp'
MLE.cpp:(.text+0x1675): undefined reference to`gsl_linalg_LU_invert'
MLE.cpp:(.text+0x16df): undefined reference to `gsl_matrix_get'
MLE.cpp:(.text+0x16f4): undefined reference to`gsl_matrix_get'
MLE.cpp:(.text+0x1706): undefined reference to `gsl_matrix_get'
MLE.cpp:(.text+0x17b5): undefined reference to`gsl_matrix_get'
MLE.cpp:(.text+0x17cd): undefined reference to `gsl_matrix_get'
MLE.o:MLE.cpp:(.text+0x17e2): more undefined references to`gsl_matrix_get' follow
MLE.o: In function `MLE_weights(int, double**, double_, double, double__)':
MLE.cpp:(.text+0x1c1e): undefined reference to `gsl_matrix_alloc'
MLE.cpp:(.text+0x1c77): undefined reference to`gsl_matrix_set'
MLE.cpp:(.text+0x1c98): undefined reference to `gsl_matrix_alloc'
MLE.cpp:(.text+0x1caa): undefined reference to`gsl_permutation_alloc'
MLE.cpp:(.text+0x1cbf): undefined reference to `gsl_linalg_LU_decomp'
MLE.cpp:(.text+0x1cd2): undefined reference to`gsl_linalg_LU_invert'
MLE.cpp:(.text+0x1d19): undefined reference to `gsl_matrix_get'
MLE.o: In function`MLE_BF_simple(int, double_, double_, double__, double, double_, double_, double&, double&, double&, double_)':
MLE.cpp:(.text+0x205a): undefined reference to `gsl_matrix_alloc'
MLE.cpp:(.text+0x20b8): undefined reference to`gsl_matrix_set'
MLE.cpp:(.text+0x20d7): undefined reference to `gsl_matrix_alloc'
MLE.cpp:(.text+0x20e9): undefined reference to`gsl_permutation_alloc'
MLE.cpp:(.text+0x20fe): undefined reference to `gsl_linalg_LU_decomp'
MLE.cpp:(.text+0x2114): undefined reference to`gsl_linalg_LU_invert'
MLE.cpp:(.text+0x2169): undefined reference to `gsl_matrix_get'
collect2: error: ld returned 1 exit status
make: *** [MV_bulkflow] Error 1
`

Could you please provide instructions which files are needed?

**Comment by mscrim** [comment]
Hi! I think you just need to install GSL: http://www.gnu.org/software/gsl/

**Comment by ufeindt** [comment]
Yes, I tried that but it didn't help. A colleague pointed out to me that g++ can be picky regarding the order of of the flag and indeed moving the LFLAGS to the end of the command solved the problem. The compiles and runs now.

I committed a version that works for me to my fork: https://github.com/ufeindt/MVBulkFlow
Since the issue may be OS-specific, I don't know whether you want to merge that. It's probably not necessary.
